winpr/libwinpr/synch/mutex.c

```diff
@@ -57,12 +65,53 @@ static int MutexGetFd(HANDLE handle)
 BOOL MutexCloseHandle(HANDLE handle)
 {
 	WINPR_MUTEX* mutex = (WINPR_MUTEX*) handle;
+	int rc;
 
 	if (!MutexIsHandled(handle))
 		return FALSE;
 
-	if (!pthread_mutex_destroy(&mutex->mutex))
+	rc = pthread_mutex_trylock(&mutex->mutex);
+	switch(rc)
+	{
+		/* If we already own the mutex consider it a success. */
+		case EDEADLK:
```

Comment:
You need to add a case EBUSY which also indicates that it is already locked.
Try running ```./Testing/TestSynch TestSynchMutex``` as proof.